gradle.properties

```diff
@@ -22,7 +22,7 @@ platformVersion = 2024.3
 platformPlugins = com.intellij.java, org.jetbrains.kotlin, org.jetbrains.idea.maven, com.intellij.gradle
 
 # Java language level used to compile sources and to generate the files for - Java 17 is required since 2023.1
-javaVersion = 17
+javaVersion = 21
```

Comment:
If it is necessary to updater the Java version, it means all our actions needs to be double-checked and potentially updated. For example, [here](https://github.com/JetBrains-Research/TestSpark/blob/c6957e569f5ad85c7cdca81925b07cf2de3126c3/.github/workflows/build.yml#L60).